### ISSUE TITLE
Working with multiple chains

### ISSUE THREAD

**Issue by adrianbrink** [tendermint/tendermint#434]
Proposed feature: 
Allow the tendermint node to work with multiple chains without having to hard reset the node.

Detail:
The node should store chain data on a per-chain basis and have flags that get interpreted at runtime to consider which chain to use. 
Another solution could be to document how to work with multiple chains with an explanation of how to backup the data. This backing up process should ideally be automated using a command on tendermint.

**Comment by melekes** [comment]
Yeah, I see a lot of people coming to the Slack having problems because they forgot to reset the validator.

Right now you can have multiple chains by setting `TMROOT` ENV variable:

```
TMROOT="~/tendermint/chain1/" tendermint node ...
TMROOT="~/tendermint/chain2/" tendermint node ...
```

There also will be a `--data` flag #431.

The question whenever we should drop the default `~/.tendermint` is debatable. I am leaning towards having a default. I don't think many people will be running multiple chains on one node. But I may be wrong.

**Comment by adrianbrink** [comment]
I think having a default is a good idea.

I think the problem with the current approach is that when you go through the tutorials you always have to reset the chains. 

Maybe an approach where you can name a chain, much like virtualenv ```--name``` would be worthwhile to explore.

**Comment by ebuchman** [comment]
Is it a problem that the tutorials require a reset? Would it be better if we were explicit about setting a new `TMROOT` each time? Or would a `--name` or `--data` flag be better?

**Comment by adrianbrink** [comment]
It makes the tutorials feel very hacky, in the sense that I feel like the software is not ready, since I have to type ```**unsafe_reset**```.

I think the TMROOT is a less elegant solution, since dealing with env variables is always a hassle to me, since it it requires a decent understanding of how your shell works and how it is initialised. 

I think the ```--name``` or ```--data``` flags are the most user-friendly versions.

**Comment by adrianbrink** [comment]
Closed in favour of #431
